### DISCUSSION TITLE
Formatting help

### DISCUSSION THREAD

**Discussion by poiromaniax** [splunk/slack-alerts#21] [Q&A] [answered]
I would love to enhance the formatting on my Splunk alerts,

for example, how to achieve this image (thats on the repo main page)
![image](https://user-images.githubusercontent.com/15910083/114381844-4af57f00-9b94-11eb-94c2-873f8c843904.png)

I would love to be able to include that quoted bottom section, but cant figure out how to do this with markdown formatting in Splunk?

**Comment by ziegfried** [comment, marked as answer]
Whatever you enter in the "Message" field in the alert follows Slack's text formatting rules. The quoted section in the screenshot is a triple-fenced codeblock (using 3 backticks). 

The example from the screenshot would look something like this:

    Suspicious authentication event detected :fire: <https://foo.com|Create incident>
    
    ```
    $result._raw$
    ```

You can use Slack's message builder to [take a look at the preview](https://api.slack.com/docs/messages/builder?msg=%7B%22text%22%3A%22Suspicious%20authentication%20event%20detected%20%3Afire%3A%20%3Chttps%3A%2F%2Ffoo.com%7CCreate%20incident%3E%5Cn%5Cn%60%60%60%5Cn%24result._raw%24%5Cn%60%60%60%5Cn%22%7D). The "message" input basically supplies the `text` property in the JSON payload. Note that Splunk would replace `$result._raw$` with the field value before sending the message to the Slack API. 

Here's what it would look like to enter the formatted message in the alert UI:
![Screenshot-20210412-113219](https://user-images.githubusercontent.com/64849/114444047-470d3100-9b83-11eb-8d70-fef21c5c45b5.png)

You can learn more about formatting slack messages here: https://api.slack.com/reference/surfaces/formatting

**Reply by poiromaniax** [reply]
Thanks! 

I will play around a bit - more trying to figure out how you did the layout such that "host" and "clientip" are on the same line under different "columns"

**Reply by ziegfried** [reply]
> trying to figure out how you did the layout such that "host" and "clientip" are on the same line under different "columns"

Those are just **fields** that I specified, special formatting needed.

![Screenshot-20210413-145725](https://user-images.githubusercontent.com/64849/114626452-97aa8a00-9c68-11eb-91eb-3f9437fba2e6.png)

Those fields need to be in the search results.

**Comment by poiromaniax** [comment]
Ahhhh, got it!

Thanks so much - very helpful. 
I never understood what the "Fields" option was about
